### PR TITLE
update parsing to account for range

### PR DESCRIPTION
**Pull request by ros3bud**
This can't be merged until the actual mortgage.amount.loan is updated in the types repo.

### CODE REVIEW

**Comment by ros3bud** [comment]
@alexkwolfe, LGTY?

File: src/outbound.coffee [hunk 1]
```diff
@@ -230,14 +230,14 @@ inRangeNum = (value, min, max) ->
...
-  return "860"    if inRangeNum(value, 500001, 1000000)
-  return "769"    if value >= 1000001
+  return "734"    if value.max <= 5000
+  return "104"    if value.min >= 5001 and value.max <= 25000
+  return "936"    if value.min >= 25001 and value.max <= 50000
+  return "938"    if value.min >= 50001 and value.max <= 100000
+  return "107"    if value.min >= 100001 and value.max <= 250000
+  return "858"    if value.min >= 250001 and value.max <= 500000
+  return "860"    if value.min >= 500001 and value.max <= 1000000
+  return "769"    if value.max <= 1000001
```
Comment (alexkwolfe) [inline src/outbound.coffee]: Why did you stop using `inRangeNum()`?

Comment (ros3bud) [inline src/outbound.coffee]: Now that loan.amount is a range, the ID can be calculated based on the min and max of that range. Our client will also be passing a range for that value, and the method is expecting a number. The method was created for parsing where a number falls within a range when Lendio is expecting an ID based on a range and we are dealing only with a number to figure that ID out.

Comment (alexkwolfe) [inline src/outbound.coffee]: Maybe it should be using `inRange()`? Also, if `inRangeNum` isn't going to be used anymore you should remove it from the source.

Comment (ros3bud) [inline src/outbound.coffee]: Yeah, it should be using that method. Thanks for catching that. I also took inRangeNum out.

**Comment by ros3bud** [comment]
@alexkwolfe ^^

**Comment by ros3bud** [comment]
At this point an error isn't thrown for this. More soon.

**Comment by ros3bud** [comment]
@alexkwolfe, LGTY? 

I needed to update the tests for null and bad values because they weren't actually working previously. I also added some tests for making sure the value is valid within the actual methods in addition to inRange. This is because in certain methods such as parseFinanceAmountId a number is being evaluated within it. However in other methods, such as parseFinanceAmountId, it is not necessary for the value to be valid until it reaches the inRange method. This is because either text or a range might be passed. Please let me know if that does not make sense.

File: package.json [hunk 2]
```diff
@@ -28,6 +28,6 @@
     "coffee-script": ">= 1.8.0",
     "mocha": ">= 1.21.0",
     "chai": "^1.10.0",
-    "leadconduit-types": "2.2.0"
```
Comment (cgrayson) [inline package.json]: I see these were previously locked to a specific version, but maybe that's not the best approach. Especially for `fields`, if it were a semver range, no change would be needed here.

File: spec/outbound-spec.coffee [hunk 3]
```diff
@@ -175,24 +175,27 @@ describe 'Outbound', ->
...
             type: 'donkey'
             monthly_sales: 'donkey'
-            industry: 'donkey'
+            accounts_receivable: 'donkey'
+            monthly_charges: 'donkey'
           loan:
-            amount: 'donkey'
+            has_financing: 'donkey'
             timeframe_days: 'donkey'
+            amount: 'donkey'
```
Comment (cgrayson) [inline spec/outbound-spec.coffee]: It's hard to follow these changes, but most seem to have just been reordered?

File: package.json (hunk 2 above)
Comment (ros3bud) [inline package.json]: Makes sense.

File: spec/outbound-spec.coffee [hunk 4]
```diff
@@ -206,28 +209,32 @@ describe 'Outbound', ->
       assert.isNull body.collateral_available
```
Comment (alexkwolfe) [inline spec/outbound-spec.coffee]: Test

File: src/outbound.coffee [hunk 5]
```diff
@@ -223,21 +223,16 @@ parseTimeInBusinessId = (value) ->
...
-  return "769"    if value >= 1000001
+  return null unless value? and value.valid == true
+  return "734"    if inRange(value, 0, 5001)
+  return "104"    if inRange(value, 5001, 25000)
+  return "936"    if inRange(value, 25001, 50000)
+  return "938"    if inRange(value, 50001, 100000)
+  return "107"    if inRange(value, 100001, 250000)
+  return "858"    if inRange(value, 250001, 500000)
+  return "860"    if inRange(value, 500001, 1000000)
+  return "769"    if value.max <= 1000001
```
Comment (cgrayson) [inline src/outbound.coffee]: I don't follow why this changed from `>=` to `<=`.

File: src/outbound.coffee [hunk 6]
```diff
@@ -320,12 +315,12 @@ parseTimeToFundingId = (value) ->
...
   return null unless value?
   valueRaw = value.toLowerCase().replace(/[^a-z]/gi, "")
-  return '878' if (value.min <= 499) || valueRaw.match(/poor/)
+  return '878' if inRange(value, 0, 499) || valueRaw.match(/poor/)
   return '880' if inRange(value, 500, 599) || valueRaw.match(/notsogood/)
   return '101' if inRange(value, 600, 649) || valueRaw.match(/okay/)
   return '100' if inRange(value, 650, 679) || valueRaw.match(/good/)   
   return '99'  if inRange(value, 680, 719) || valueRaw.match(/great/)
-  return '98'  if value.max >= 720 || valueRaw.match(/excellent/)
+  return '98'  if value.max >= 720 || valueRaw.match(/^excellent/)
```
Comment (cgrayson) [inline src/outbound.coffee]: Why does "excellent" get the special regex treatment? :smiley:

Comment (ros3bud) [inline src/outbound.coffee]: It should not. Thanks for catching that. It was an experiment I meant to delete.

File: src/outbound.coffee (hunk 5 above)
Comment (ros3bud) [inline src/outbound.coffee]: Again, thanks for catching this. That was a mistake. I will attribute it to being distracted by a bee sting.

File: src/outbound.coffee [hunk 7]
```diff
@@ -249,9 +244,8 @@ parsePrimaryCustomerId = (value) ->
   return null
  
 parseAverageMonthlySalesId = (value) ->
-  return null unless value?
-  return "856"    if inRange(value, 1, 999)
-  return "1150"   if (value.min < 1)
+  return null unless value? and value.valid == true
+  return "1150"   if inRange(value, 0, 1)
   return "1152"   if inRange(value, 1, 4000)
```
Comment (cgrayson) [inline src/outbound.coffee]: 1. "856" is no longer a value to return, is that right? 
2. The presence of "1" in both of these lines makes me wonder if they're both right.

**Comment by cgrayson** [comment]
Other than those questions, LGTM

File: src/outbound.coffee (hunk 7 above)
Comment (ros3bud) [inline src/outbound.coffee]: Correct on "856". 
I was trying to make sure that anything less than 1 would return 1150, but after looking at the docs again, I think that 1150 should only be returned when the average monthly sales in 0. Anything more than 0 should be 1152.
